src/epoll.rs

```diff
@@ -225,6 +228,20 @@ impl Poller {
     }
 }
 
+impl AsRawFd for Poller {
+    fn as_raw_fd(&self) -> RawFd {
+        self.epoll_fd
+    }
+}
+
+#[cfg(not(polling_no_io_safety))]
```

Comment:
nit: double-negative makes it harder to read (I first misread it). I'd suggest:

```rust
#[cfg(polling_io_safety)]
```

(I also wouldn't use `polling_` prefix but no strong feelings there so just a suggestion :smile:)

Reply:
This is intentional, see https://github.com/smol-rs/async-io/pull/84#discussion_r948085917

Reply:
Also, [this comment](https://github.com/tokio-rs/tokio/blob/4b1c4801b1383800932141d0f6508d5b3003323e/tokio/build.rs#L74-L78) describes another case where `no_*` cfg is reasonable.

Reply:
> This is intentional, see https://github.com/smol-rs/async-io/pull/84#discussion_r948085917 for the context.

Thanks but why do we want to assume latest rust if build script is not run? Also, why is build script not run?

I'd be a least good to have [this comment](https://github.com/tokio-rs/slab/blob/817a2ec227e819f308e8e4d0487084940b4ef831/build.rs#L15-L20) here as well so it's clear to the reader, not just me as a reviewer (after asking for it :)).

Reply:
I added the comment. There are non-`cargo` build tools that don't run `build.rs` for various reasons. For instance, I can't remember which project it is, but they build some Rust code in a C/C++ codebase by running `rustc` directly through a Makefile. 

In this case, since we have no versioning information available to us, we make the (reasonable) assumption that we're just running the latest stable version. Since `build.rs` is never run, `polling_no_io_safety` is never set.

Reply:
> I added the comment. There are non-`cargo` build tools that don't run `build.rs` for various reasons. For instance, I can't remember which project it is, but they build some Rust code in a C/C++ codebase by running `rustc` directly through a Makefile.

Thanks for explaining! This a Cargo-based project and I really don't see why we should sacrifice the readability of this project for possible users that use cargo-based projects w/o ensuring their build system is actually doing everything cargo-based project expects. However, now that you've added the comment, it's nothing I'd block your good work for. :)